src/DynamoCore/Utilities/LuceneSearchUtility.cs

```diff
@@ -84,25 +127,37 @@ internal void InitializeLuceneConfig(string dirName, LuceneStorage storageType =
             // Create an analyzer to process the text
             Analyzer = CreateAnalyzerByLanguage(dynamoModel.PreferenceSettings.Locale);
 
+            // Check if Lucene index file exists, if not create it
+            if (!DirectoryReader.IndexExists(indexDir))
+            {
+                CreateLuceneIndex();    
+            }
+        }
+
+        internal void CreateLuceneIndex()
+        {
             // Initialize Lucene index writer, unless in test mode or we are using RAMDirectory for indexing info. 
-            if (!DynamoModel.IsTestMode || currentStorageType == LuceneStorage.RAM)
+            if (!DynamoModel.IsTestMode || startConfig.StorageType == LuceneStorage.RAM)
             {
+                // Create an index writer
+                IndexWriterConfig indexConfig = new IndexWriterConfig(LuceneConfig.LuceneNetVersion, Analyzer)
+                {
+                    OpenMode = OpenMode.CREATE
+                };
                 try
                 {
-                    // Create an index writer
-                    IndexWriterConfig indexConfig = new IndexWriterConfig(LuceneConfig.LuceneNetVersion, Analyzer)
-                    {
-                        OpenMode = OpenMode.CREATE
-                    };
-
                     writer = new IndexWriter(indexDir, indexConfig);
+                    writer.Commit();
                 }
                 catch (LockObtainFailedException ex)
                 {
-                    DisposeWriter();
-                    dynamoModel.Logger.LogError($"LuceneNET LockObtainFailedException {ex}");
+                    writer = new IndexWriter(new RAMDirectory(), indexConfig);
+                    writer.Commit();
+                    // DisposeWriter();
```

Comment:
??

Reply:
This is done so that in second Dynamo session when the file is locked, to switch to RAM mode instead of just giving up